### DISCUSSION TITLE
Why should you use Reusable Workflows?

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#33041] [Discover: GitHub Best Practices]
With almost a decade into my professional journey, I’ve seen developers across the globe progressively seeking new ways to improve productivity while programming, and have efficient development workflows. GitHub is increasingly becoming a favourite centralized platform where developers code and deploy software. GitHub Actions is the one-stop solution for today’s fast paced CI/CD integrations that allows developers to automate, customize, and execute their workflows from the repository itself. 🚀 
 
Do you often write GitHub Actions workflows to accelerate your application development? Have you ever gotten stuck or confused in a situation where you had to re-write your [Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) workflow and just couldn’t find the right way to reuse them? 🤔 

Well, not anymore! GitHub offers [Reusable Workflows](https://github.blog/changelog/2021-11-24-github-actions-reusable-workflows-are-generally-available/), which spares you from writing the same workflow(s) again and again, and instead, enables re-using GitHub Action workflows as easy as pie. ♻️ 

![Screenshot 2022-09-14 at 6 29 02 PM](https://user-images.githubusercontent.com/95834130/190160581-7e2b0c7a-3e35-49af-b94b-2f1a677b6008.png)

Reusable workflows are GitHub Actions templates on steroids! In simple terms, they allow you to reference an entire workflow from a different GitHub Actions workflow, with a single line of configuration, rather than copying and pasting from one workflow to another. It makes maintaining workflows easy and allows you to create new ones quickly, by building on the work of others. ❤️ 

In order to get triggered, workflows must be in the .github/workflows folder. Reusable Workflows listen for workflow_call, a special event that needs access to all repositories in your organization. Here’s an example of Reusable workflow that takes an input string along with a secret from the caller workflow so that it can be reused in an action:

```
name: Call a reusable workflow
on:
  workflow_dispatch:
jobs:
  reusable-job:
    uses: <owner>/<repo>/.github/workflows/<reusable workflow>@master
    with:
      username: GitHub
    secrets:
      token: test
```
### Why should you use Reusable Workflows? 💭 

- Reduces redundancy: If you have multiple repositories deployed the same way, reusable workflows can help you keep them in sync.
- No Duplication: We already know that by referencing workflows in other GitHub Action workflow, you reproduce the same work smartly! 
- Easy to create: All you need to have is a trigger and a workflow_call to prompt it. This simple and effortless process is explained [here](https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow).

Now that you know why Reusable Workflow is right for you, read these [three scenarios](https://github.blog/2021-11-29-github-actions-reusable-workflows-is-generally-available/) to know how Reusable Workflows can simplify your GitHub Actions scalability. If you’d like to take a deep dive into Reusable Workflows with real life examples, I highly recommend that you checkout this [post](https://github.blog/2022-02-10-using-reusable-workflows-github-actions/) from the GitHub blogs. For questions, please visit the [GitHub Actions community forum](https://github.community/c/code-to-cloud/github-actions).

### Are you new to GitHub Actions? 🆕 

- Get started with learning GitHub Actions with just 3 easy steps. Checkout these [pro-tips](https://github.com/orgs/community/discussions/30965) from [talktopri](https://github.com/talktopri) 
- Bookmark these [resources](https://github.blog/2021-11-04-10-github-actions-resources-basics-ci-cd/) for your GitHub Actions learning journey
- Explore the [GitHub Actions Documentation](https://docs.github.com/en/actions)

Want to keep track of what's coming next for Actions? ❓  [Visit our public roadmap](https://github.com/orgs/github/projects/4247/views/1?filterQuery=actions)

**Comment by To-Rex** [comment]
ajoyib
